src/hotspot/share/runtime/lockStack.cpp

```diff
@@ -99,6 +103,60 @@ void LockStack::verify(const char* msg) const {
 }
 #endif
 
+#ifdef ASSERT
+void LockStack::verify_consistent_lock_order(GrowableArray<oop>& lock_order, bool leaf_frame) const {
```

Comment:
Thanks for adding the verification code.

I spun my head around a bit with `exec_mode != Deoptimization::Unpack_none`
being passed as `leaf_frame`, but I think I got myself reoriented on the correct
meaning of `if (!leaf_frame) {`.